otel-tests/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use quote::ToTokens;
+use syn::ItemFn;
+use syn::Lit;
+use syn::MetaNameValue;
+use syn::Path;
+
+#[proc_macro_attribute]
+pub fn otel_test(attribute: TokenStream, item: TokenStream) -> TokenStream {
+    let fn_item = syn::parse::<ItemFn>(item).unwrap();
+
+    let sig = fn_item.sig;
+    let block = fn_item.block;
+    let name = &sig.ident;
+    let attrs = fn_item.attrs;
+
+    let test = if attrs.is_empty() {
+        quote!(#[tokio::test])
+    } else {
+        quote!()
+    };
+
+    let init_tracing = if attribute.is_empty() {
+        quote!(::daemon_tests::init_tracing)
```

Comment:
this crate should not depend on `daemon-tests` (especially in such a sneaky way, when it compiles and then fails on usage!)

IMHO we should move `init_tracing()` inside this crate, and also consider making it public (if someone doesn't want / need the macro)

Alternatively, we could move the proc macro into daemon-tests; that said, we might want to sometimes instrument tests e.g. like recently, the load test from xtra-libp2p - so separate crate makes sense.